### PR TITLE
Swagger closure for Azure Migration Hub

### PR DESCRIPTION
**Pull request by peetlotla**
Incorporated changes as per ARM review.

### CODE REVIEW

**Comment by ravbhatnagar** [comment]
@SDK team- this should not be merged till the service team indicates as the service side is not ready.

**Comment by ravbhatnagar** [comment]
Signed off from ARM side, this was reviewed over a meeting and all the feedback has been addressed.

File: specification/migrateprojects/resource-manager/readme.nodejs.md
Comment (daschult) [inline specification/migrateprojects/resource-manager/readme.nodejs.md]: Please also create a readme.typescript.md file. You can find several other packages in this repository that have readme.typescript.md files that you can copy and modify slightly to get the results you need.

**Comment by salameer** [comment]
@peetlotla what the status here, this PR has been opened for month and will be closed if there no update by end of week

**Comment by peetlotla** [comment]
Hi,

I am checking this with the ARM review team. Please do not close this until the next update.

Regards,
-Peet

**Comment by peetlotla** [comment]
+ Samer

From: Peetambaram Lotla
Sent: Friday, April 5, 2019 5:25 PM
To: 'Azure/azure-rest-api-specs' <reply@reply.github.com>; Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: Mention <mention@noreply.github.com>
Subject: RE: [Azure/azure-rest-api-specs] [Do not merge] Swagger closure for Azure Migration Hub (#4481)

Hi,

I am checking this with the ARM review team. Please do not close this until the next update.

Regards,
-Peet

**Comment by salameer** [comment]
Cool :)

Kindly Avoid pasting emails on the Public repo :)

**Comment by salameer** [comment]
@peetlotla any updates here? if there's an email thread with ARM loop me into it and I can help move forward

**Comment by praries880** [comment]
@peetlotla If the PR is ready for merge can you kindly update the title to reflect that?

**Comment by peetlotla** [comment]
I am still waiting on the ARM team to see if it’s OK to merge. Will update it once when I get the sign off from them.

**Comment by salameer** [comment]
@peetlotla any updates here?

**Comment by ravbhatnagar** [comment]
@peetlotla - Are there changes that need a ARM re-review? I am checking since we reviewed and signed off on this few months ago.

**Comment by peetlotla** [comment]
> @peetlotla - Are there changes that need a ARM re-review? I am checking since we reviewed and signed off on this few months ago.

@ravbhatnagar - The swagger spec has changes related to Databases integration with Azure Migrate Hub. Also, as discussed in the mail I have used the discriminator pattern to include sub classes in the swagger spec.

**Comment by peetlotla** [comment]
ARM feedback based on discussion with @ravbhatnagar:

1. Database instance id can be used as the arm name for the instance 
2. Solution name in the properties needs to be fully qualified ARM name
3. Database instances and databases can have the parent child hierarchy.

**Comment by peetlotla** [comment]
> ARM feedback based on discussion with @ravbhatnagar:
> 
> 1. Database instance id can be used as the arm name for the instance
> 2. Solution name in the properties needs to be fully qualified ARM name
> 3. Database instances and databases can have the parent child hierarchy.

Since GA is approaching we would like to go ahead and merge this PR. As discussed the feedback is not a breaking change so we need to work on this feedback post GA or whenever the next update happens for this resource.

**Comment by nschonni** [comment]
You can get the spellcheck build to pass by adding the new words to https://github.com/Azure/azure-rest-api-specs/blob/master/custom-words.txt

**Comment by ravbhatnagar** [comment]
As part of review done on 6/18, we just reviewed the delta as presented by service team. This included reviewing the two new types being added - databases and databaseInstances. 
Service team has agreed to make the changes (#1 and #2) after the GA milestone since these are non-breaking changes. #3 is still not committed but will be explored. Signing off on this change from ARM side.

**Comment by peetlotla** [comment]
@Azure/azure-rest-api-specs<mailto:reply@reply.github.com> – Can you plz help merge the PR<https://github.com/Azure/azure-rest-api-specs/pull/4481>? It’s has been signed off by ARM review team.

**Comment by peetlotla** [comment]
This is important for Azure Migrate Hub release which is time critical. Please help.

From: Peetambaram Lotla
Sent: Wednesday, June 19, 2019 9:09 PM
To: Azure/azure-rest-api-specs <reply@reply.github.com>; Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>; Samer Alameer <sameal@microsoft.com>
Cc: Mention <mention@noreply.github.com>
Subject: RE: [Azure/azure-rest-api-specs] [Do not merge] Swagger closure for Azure Migration Hub (#4481)

@Azure/azure-rest-api-specs<mailto:reply@reply.github.com> – Can you plz help merge the PR<https://github.com/Azure/azure-rest-api-specs/pull/4481>? It’s has been signed off by ARM review team.

**Review by praries880 — APPROVED** [review]

**Pull request merged**
